pandas/core/strings.py

```diff
@@ -1058,7 +1058,8 @@ class StringMethods(object):
     """
 
     def __init__(self, series):
-        self.series = series
+        # can't use direct assignment as we prevent that in __set_item__
+        object.__setattr__(self, "series", series)
```

Comment:
try making this inherit from ``PandasDelegate`` rather than have the changes twice.